### PR TITLE
fix: set Loc Line on Decl

### LINKED ISSUE #2220: We didn't set File line on ValueDeclr ImportDecl, TypeDecl
#### Description:
The following code snippet produces an error in Gnolang, but the error message is misleading and does not point to the correct line number.

#### Code to Reproduce:
```go
package main

func main() {
  var a, b, c = 1, 2
}
```

#### Expected Behavior:
The code should either compile correctly or produce a clear error message indicating the exact issue and the correct line number.

#### Actual Behavior:
The code returns the following error:
```
panic: main//src/package.gno:3: should not happen
```
The part `"main//src/package.gno:3:"` is not accurate because it does not set the line number of the `ValuDelr` type correctly. Instead, it takes the line number of the `blocknode`, which is misleading.

### PR DESCRIPTION
**Pull request by omarsy**
Closes #2220

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: gnovm/tests/files/var18.gno [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package main
+
+func main() {
+	a, b, c := 1, 2
+}
+
+// Error:
+// main/files/var18.gno:4: should not happen
```
Comment (ltzmaxwell) [inline gnovm/tests/files/var18.gno:8]: let's change this to be  something descriptive, thank you.

Comment (omarsy) [reply gnovm/tests/files/var18.gno:8]: Yes I will do it this PR: #2206

Maybe I should remove this example, in this PR ?

Comment (ltzmaxwell) [reply gnovm/tests/files/var18.gno:8]: OK. I think you can leave it as a show case of  `correct loc` info.

File: gnovm/tests/files/const9.gno [hunk 2]
```diff
@@ -14,4 +14,4 @@ func main() {
 }
 
 // Error:
-// main/files/const9.gno:1: constant definition loop with b
```
Comment (thehowl) [inline gnovm/tests/files/const9.gno:17]: This is incorrect: it's considering :3 as the location instead of `c` (or wherever else this happens)

Comment (omarsy) [reply gnovm/tests/files/const9.gno:17]: It is considered line 3 as the starting point for the declaration statement that begins with `const (` and ends at line 10. I will investigate if we can pinpoint the exact assignment line with the current design.

Comment (omarsy) [reply gnovm/tests/files/const9.gno:17]: It is possible to replicate the cycle lines like Golang's as follows (to make the error more accurate):

```
./prog.go:5:2: initialization cycle for b
	./prog.go:5:2: b is referred to
	./prog.go:6:2: c is referred to
	./prog.go:7:2: d is referred to
	./prog.go:8:2: e is referred to
	./prog.go:5:2: b
```
However, some refactoring is required in the functions [predefinenow](https://github.com/gnolang/gno/blob/master/gnovm/pkg/gnolang/preprocess.go#L3001) and [trydefine](https://github.com/gnolang/gno/blob/master/gnovm/pkg/gnolang/preprocess.go#L3154) to make this error more precise.

Should we implement these changes in this PR?

Comment (ltzmaxwell) [reply gnovm/tests/files/const9.gno:17]: I guess you want introduce a dependency graph for this, but i’m not sure we really should do it. had a similar talk in #2081.

Comment (omarsy) [reply gnovm/tests/files/const9.gno:17]: I plan to use the existing check circularity variable at https://github.com/gnolang/gno/blob/master/gnovm/pkg/gnolang/preprocess.go#L3036-L3039. If we encounter the first circularity, I will handle the panic situation differently I  want to  display all variables concerned.

Comment (thehowl) [reply gnovm/tests/files/const9.gno:17]: @omarsy please, tackle this in another PR: the issue is not circularity, but where we mark the error as happening. the location should happen at the individual declaration, not at the start of the declaration block. thanks.

Comment (omarsy) [reply gnovm/tests/files/const9.gno:17]: @thehowl, is this one good for you?
Commit: https://github.com/gnolang/gno/pull/2221/commits/eaab6e11e1eea0b39a2f84ac3dcad20afe7f4429 ?
I no longer use the line from the declaration group.

**Review by thehowl — APPROVED** [review]
:ok_hand: 

cc @mvertes to verify the changes on the debugger outputs.

**Review by jaekwon — APPROVED** [review]

**Pull request merged**